app/src/main/java/com/ifeins/tenbismonit/fragments/OverviewFragment.java

```diff
@@ -83,41 +96,36 @@ private void updateUI(@Nullable DocumentSnapshot document) {
         mErrorView.setVisibility(View.GONE);
 
         if (document == null || !document.exists()) {
+            mRecyclerView.setVisibility(View.GONE);
             mProgressBarView.setVisibility(View.VISIBLE);
             mProgressCaptionView.setVisibility(View.VISIBLE);
-            mImageView.setVisibility(View.GONE);
-            mBudgetView.setVisibility(View.GONE);
-            mLunchesView.setVisibility(View.GONE);
-            mTotalSpentView.setVisibility(View.GONE);
-            mAverageLunchView.setVisibility(View.GONE);
-            mTodayBudgetView.setVisibility(View.GONE);
-            mLastUpdateView.setVisibility(View.GONE);
             ((HomeActivity) getActivity()).syncUserData();
             return;
         }
 
         String updatedAt = document.getString("updatedAt");
 
+        mRecyclerView.setVisibility(View.VISIBLE);
         mProgressBarView.setVisibility(View.GONE);
         mProgressCaptionView.setVisibility(View.GONE);
-        mImageView.setVisibility(View.VISIBLE);
-        mBudgetView.setVisibility(View.VISIBLE);
-        mLunchesView.setVisibility(View.VISIBLE);
-        mTotalSpentView.setVisibility(View.VISIBLE);
-        mAverageLunchView.setVisibility(View.VISIBLE);
-        mTodayBudgetView.setVisibility(View.VISIBLE);
-        mLastUpdateView.setVisibility(TextUtils.isEmpty(updatedAt) ? View.GONE : View.VISIBLE);
-
-        mBudgetView.setText(getString(R.string.remaining_budget,
-                document.get("remainingMonthlyLunchBudget"), document.get("monthlyLunchBudget")));
-        mLunchesView.setText(getString(R.string.remaining_lunches,
-                document.get("remainingLunches"), document.get("workDays")));
-        mTotalSpentView.setText(getString(R.string.total_spent,
-                document.get("totalSpent")));
-        mAverageLunchView.setText(getString(R.string.average_lunch_spending,
-                document.get("averageLunchSpending")));
-        mTodayBudgetView.setText(getString(R.string.today_budget,
-                document.get("todayBudget")));
+
+        ArrayList<CardData> data = new ArrayList<>();
+        String remainingBudgetString = getString(R.string.remaining_budget_value, document.get("remainingMonthlyLunchBudget"), document.get("monthlyLunchBudget"));
+        SpannableString remainingBudget = new SpannableString(remainingBudgetString);
+        remainingBudget.setSpan(new RelativeSizeSpan(0.7f), remainingBudgetString.indexOf("\n"), remainingBudgetString.length(), 0);
```

Comment:
That's kinda weird, better to extract it to `remaining_budget_header_string` and `remaining_budget_value_string`